File: server/models/category.py
from typing import List
from pydantic import BaseModel, Field
from enum import Enum


class StatusGB(Enum):
    END = 0
    IC = 1
    LIVE = 2


class CategoryType(Enum):
    Keeb = 0
    Keycap = 1
    ETC = 2


class Image(BaseModel):
    path: str
    size: float


class CategorySchema(BaseModel):
    category_id: str = Field(...)
    category_url_name: str
    category_name: str
    manufacturing: str
    author: str
    proxy_host: str
    status_gb: StatusGB
    type: CategoryType
    date_start: str
    date_end: str
    date_payment: str
    min_price: int
    max_price: int
    tax: int
    handle: int
    thumbnail: Image
    image_list: List[Image]
    content: str
    time_line: None

    class Config:
        schema_extra = {
            "example": {
                "category_id": 'abc',
                "category_url_name": "123456abc",
                "category_name": "Hari",
                "manufacturing": "Chan",
                "author": "A Lu",
                "proxy_host": "text",
                "status_gb": 0,
                "type": 0,
                "date_start": '',
                "date_end": '',
                "date_payment": '',
                "min_price": 0,
                "max_price": 0,
                "tax": 0,
                "handle": 0,
                "thumbnail": {
                    "path": "",
                    "size": 0
                },
                "image_list": [{"paht": "", "size": 0}],
                "content":  "",
                "get_noti": False,
                "time_line": {}
            }
        }


def ResponseModel(data, message):
    return {
        "data": [data],
        "code": 200,
        "message": message
    }


def ErrorResponseModel(error, code, message):
    return {"error": error, "code": code, "message": message}

File: databaseModel/category.js
var mongoose = require('mongoose');
var AutoIncrement = require('mongoose-sequence')(mongoose);
var date = new Date();



var categorySchema = mongoose.Schema({
    count:{
        type: Number
    },
    category_id:{
        type: String,
        uppercase: true,
        require: true,
    },
    category_url_name:{
        type: String,
        lowercase: true
    },
    category_name:{
        type: String
    },
    manufacturing:{
        type: String
    },
    author:{
        type: String
       
    },
    proxy_host:{
        type: String
    },
    status_gb:{ // status: 0 = end | 1 = IC | 2 = live
        type: Number,
        default: 1
    },
    k_color: [],
    
    c_code_color:[], 
    type:{ // type: 0 = keeb | 1 = keycap | 2 = etc
        type: Number,
    },
    c_profile:{
        type: String
    },
    c_material:{
        type: String
    },
    k_layout:[]
    // {
    //     type: String,
    //     default: null
    // }
    ,
    flip:{
        // type: Boolean,
        // default: false
        type: Number
    },
    k_degree:[],
    k_mounting:[],
    k_plate_option:[],
    sale_type:[],
    date_start:{
        type: Date,
        default: null,
    },
    date_end:{
        type: Date,
        default: null
    },
    date_payment:{
        type: Date,
        default: null
    },
    min_price:{
        type: Number,
        default: null
    },
    max_price:{
        type: Number,
        default: null
    },
    tax:{
        type: Number,
        default: null
    },
    handle:{
        type: Number,
        default: null
    },
    pic_profile:
    {
        path: {
            type: String
        },
        size: {
            type: Number
        },
        
    },
    pic_list: [
        {
            path: {
                type: String
            },
            size: {
                type: Number
            },
            
        }
    ],
    product_belong:{
        type: String,
        default: null,
    },
    summary_content:{
        type: String
    },
    price_content:{
        type: String
    },
    specs:{
        type: String
    },
    time_line:{
        type: String
    },
    category_intro:{
        type: String
    }

})

categorySchema.plugin(AutoIncrement, {id : 'category_seq', inc_field: 'count'});



let Category = mongoose.model('Category', categorySchema);
module.exports = Category;